src/main/resources/com/google/api/codegen/py/discovery_fragment.snip

```diff
@@ -8,51 +8,93 @@
-         serviceTitle = context.getService.getTitle
+         serviceTitle = context.getService.getTitle, \
+         authType = context.getApiaryConfig.getAuthType.toString
         """
         BEFORE RUNNING:
         ---------------
         1. If not already done, enable the {@serviceTitle}
            and check the quota for your project at
            https://console.developers.google.com/apis/api/{@apiName}
-        2. This sample uses Application Default Credentials for authentication.
-           If not already done, install the gcloud CLI from
-           https://cloud.google.com/sdk/ and run
-           'gcloud beta auth application-default login'
-        3. Install the Python client library for Google APIs by running
-           'pip install --upgrade google-api-python-client'
+        @if authType == "APPLICATION_DEFAULT_CREDENTIALS"
+          2. This sample uses Application Default Credentials for authentication.
+             If not already done, install the gcloud CLI from
+             https://cloud.google.com/sdk/ and run
+             'gcloud beta auth application-default login'
+          3. Install the Python client library for Google APIs by running
+             'pip install --upgrade google-api-python-client'
+        @else
+          2. Install the Python client library for Google APIs by running
+             'pip install --upgrade google-api-python-client'
+        @end
         """
 
         @join import : imports
             {@import}
         @end
+        {@body}
+    @end
+@end
 
+@private adcBoilerplate()
         @# Authentication is provided by the 'gcloud' tool when running locally
         @# and by built-in service accounts when running on GAE, GCE, or GKE.
         @# See https://developers.google.com/identity/protocols/application-default-credentials for more information.
         credentials = GoogleCredentials.get_application_default()
-
-        @# Construct the {@apiName} service object (version {@apiVersion}) for interacting
-        @# with the API. You can browse other available API services and versions at
-        @# https://developers.google.com/api-client-library/python/apis/
-        {@serviceName()} = discovery.build('{@apiName}', '{@apiVersion}', credentials=credentials)
-        {@body}
-    @end
 @end
 
 @snippet generateBody(method)
     {@alwaysImport()}
-    @let params = context.getApiaryConfig.getMethodParams(method.getName)
-        {@initialize(method, params)}
-
-        @if context.hasRequestField(method)
-            {@initializeRequest(method)}
+    @let api = context.getApi, \
+         apiName = api.getName, \
+         apiVersion = api.getVersion, \
+         serviceTitle = context.getService.getTitle, \
+         apiaryConfig = context.getApiaryConfig, \
+         authType = apiaryConfig.getAuthType.toString
+      @switch authType
+      @case "APPLICATION_DEFAULT_CREDENTIALS"
+        {@importGoogleCredential()}{@adcBoilerplate()}
+
+      @case "OAUTH_3L"
```

Comment:
In this case it's because there's no boilerplate required for the API_KEY flow, unlike Java.
All that's needed is for the service to be constructed with the `developerKey` option.